pyfstat/mcmc_based_searches.py

```diff
@@ -3322,40 +3318,34 @@ def _initiate_search_object(self):
             self.maxStartTime = self.search.maxStartTime
 
     def logl(self, theta, search):
-        for j, theta_i in enumerate(self.theta_idxs):
-            self.fixed_theta[theta_i] = theta[j]
-        in_theta = copy.copy(self.fixed_theta)
-        in_theta[1] = in_theta[0] + in_theta[1]
-        if in_theta[1] > self.maxStartTime:
+        in_theta = {
+            key: self.fixed_theta[k]
+            for k, key in enumerate(self.full_theta_keys)
+            if "transient" not in key
+        }
+        in_theta.update(
+            {
+                key: theta[k]
+                for k, key in enumerate(self.theta_keys)
+                if "transient" not in key
+            }
+        )
+        # FIXME: this can be simplified when changing all theta lists to dicts
+        t0idx = self.theta_keys.index("transient_tstart")
+        t1idx = self.theta_keys.index("transient_duration")
+        in_theta["tstart"] = theta[t0idx]
+        in_theta["tend"] = theta[t0idx] + theta[t1idx]  # endtime = t0+tau
+        if in_theta["tend"] > self.maxStartTime:
             return -np.inf
-        twoF = search.get_fullycoherent_twoF(*in_theta)
+        twoF = search.get_fullycoherent_twoF(**in_theta)
```

Comment:
Ideally this is how I think the function should always be called, and a follow-up PR could change the whole MCMC module to always use dicts and no lists for all its `theta` needs and then do it like here.

Reply:
Agree